File: app/src/main/java/br/com/amilton/viewmodel/StatementViewModel.java
package br.com.amilton.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import br.com.amilton.model.Statement;
import br.com.amilton.model.UserAccount;
import br.com.amilton.repository.StatementRepository;

public class StatementViewModel extends ViewModel {

    private final MutableLiveData<UserAccount> userAccountMutableLiveData = new MutableLiveData<>();
    private final StatementRepository statementRepository;

    public StatementViewModel() {
        statementRepository = new StatementRepository();
    }

    public LiveData<List<Statement>> getStatementsMutableLiveData(String userId) {
        return statementRepository.getStatements(userId);
    }

    public MutableLiveData<UserAccount> getUserAccount() {
        return userAccountMutableLiveData;
    }

    public void setUserAccount(UserAccount userAccount) {
        userAccountMutableLiveData.setValue(userAccount);
    }

}

File: app/src/main/java/br/com/flaviokreis/santanderv2/viewmodels/StatementViewModel.kt
package br.com.flaviokreis.santanderv2.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import br.com.flaviokreis.santanderv2.data.model.UserAccount
import br.com.flaviokreis.santanderv2.data.repositories.LoginRepository
import br.com.flaviokreis.santanderv2.data.repositories.StatementRepository
import br.com.flaviokreis.santanderv2.data.response.StatementsResponse
import javax.inject.Inject

class StatementViewModel @Inject constructor(
    private val loginRepository: LoginRepository,
    private val repository: StatementRepository
) : ViewModel() {

    fun getStatements(): LiveData<StatementsResponse> {
        return repository.getStatements()
    }

    val userAccount : LiveData<UserAccount>
        get() = loginRepository.getUserAccount()

    fun hasUserAccount(): LiveData<Boolean> {
        return loginRepository.hasUserAccount()
    }

    fun logout() {
        loginRepository.logout()
    }
}